File: Python/kth_elem_BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        stack = []
        def helper(node, stack):
            if node:
                stack.append(node.val)
            if node.left:
                helper(node.left, stack)
            if node.right:
                helper(node.right, stack)
        helper(root, stack)
        stack.sort()
        # print(stack)
        return stack[k-1]

File: Cpp/kth_elem_BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> stack;
    int kthSmallest(TreeNode* root, int k) {
        helper(root);
        sort(stack.begin(), stack.end()); 
        return stack[k-1];
    }

    void helper(TreeNode* node){
        if (node != NULL) stack.push_back(node->val);
        if (node->left != NULL) helper(node->left);
        if (node->right != NULL) helper(node->right);
    }
};